File: primeseive.py
#finds all primes from 2 to bound, returns a list
from math import sqrt, ceil
import timeit, numpy as np
start = timeit.default_timer()

def seive(bound):
  boolArr = np.zeros(bound)
  primeArr = []
  boolArr[0] = False; boolArr[1] = False#seive starts at 2, and 0 and 1 are not prime. 

  for i in range(2,bound): 
    boolArr[i] = True#sets all to true

  for a in range(2,ceil(sqrt(bound))+1):
    if boolArr[i]:
      for j in range(a*a, bound, a):
        boolArr[j] = False#sets non primes to false
  
  for p in range(len(boolArr)):
    if boolArr[p] == True:
      primeArr.append(p)

  return primeArr

print(seive(20))

stop = timeit.default_timer()
print('Time: ',stop-start)

File: primeseive.swift
//prime seive: generates all prime numbers <bound
import Foundation

func seive(bound: Int) -> Array<Int>
{
    var boolArr = Array (repeating: true, count: bound)
    var primeArr = [Int]()
    boolArr[0] = false; boolArr[1] = false;
    
    for a in 2...(Int(ceil(sqrt(Double(bound))))+1) {
        if boolArr[a] {
            for j in stride(from: a*a, to: bound, by: a) {
                boolArr[j] = false
            }
        }
    }
    for p in 0...boolArr.count-1 {
        if boolArr[p] == true {
            primeArr.append(p)
        }
    }
    return (primeArr)
}

print(seive(bound:1000))
